[DOCS] 7.17.6 & 7.17.5 Windows Endpoints may wind up in a non-running state (#2492)
elastic/security-docs: docs/release-notes/7.17.asciidoc

[[release-notes-7.17.6]]
=== 7.17.6

[discrete]
[[known-issue-7.17.6]]
==== Known issues

> * In some situations, {elastic-endpoint} might change to a non-running state on Windows endpoints and cannot be restarted. {agent} will have an `Unhealthy` status when this happens (https://github.com/elastic/endpoint/issues/29[#29]).

Reviewer: ```suggestion
* In some situations, {elastic-endpoint} might change to a non-running state on Windows endpoints and fail to restart. {agent} will have an `Unhealthy` status when this happens (https://github.com/elastic/endpoint/issues/29[#29]).
```

Reviewer: if you accept this suggestion you can apply it to the other version of release notes in this PR as well

+
To determine whether {elastic-endpoint} has stopped running because of this issue, run the following PowerShell command as an administrator:

+
[source,console]
--------------------------------------------------
PS C:\Users\user> Get-WinEvent Microsoft-Windows-CodeIntegrity/Operational | where Id -eq 3004 | where Message -match "elastic-endpoint.exe"

   ProviderName: Microsoft-Windows-CodeIntegrity

TimeCreated                      Id LevelDisplayName Message
-----------                      -- ---------------- -------
9/22/2022 10:47:35 AM          3004 Error            Windows is unable to verify the image integrity of the file \Device\HarddiskVolume3\Program Files\Elastic\Endpoint\elastic-endpo...
9/19/2022 2:10:14 PM           3004 Error            Windows is unable to verify the image integrity of the file \Device\HarddiskVolume3\Program Files\Elastic\Endpoint\elastic-endpo...
--------------------------------------------------

+

If {elastic-endpoint} is not running, there are several workarounds you can take:

> ** **Trigger an Endpoint reinstall by manually uninstalling {elastic-endpoint} on affected hosts**: Remove an invalid {elastic-endpoint} install by manually invoking the {elastic-endpoint} <<uninstall-endpoint,uninstall command>> on affected hosts. Once the uninstallation process has finished, use the command below to restart {agent}. Restarting {agent} automatically begins {elastic-endpoint}'s re-installation process:

Reviewer 2: Kind of a silly question -- but what do we mean by `"manually invoking the {elastic-endpoint} <<uninstall-endpoint,uninstall command>> on affected hosts"`?  I understand this to mean simply mean run the uninstall endpoint command. If so, we should simplify the language -- but wanted to check to make sure I understood first. 

Alt suggestion: 
`Remove an invalid {elastic-endpoint} install by running the {elastic-endpoint} <<uninstall-endpoint,uninstall command>> on affected hosts.`

```suggestion
** **Begin an endpoint reinstall by manually uninstalling {elastic-endpoint} on affected hosts**: Remove an invalid {elastic-endpoint} install by manually invoking the {elastic-endpoint} <<uninstall-endpoint,uninstall command>> on affected hosts. Once the uninstallation process has finished, run the following command to restart {agent}, which automatically starts {elastic-endpoint}'s re-installation process:
```

Reviewer 2: Thanks @joepeeples. One small thing -- I believe `reinstall` is sans the hyphen. Example in Elasticsearch [here](https://www.elastic.co/guide/en/elasticsearch/reference/8.4/zip-windows.html).

Author: Hey, @jmikell821 ! I was actually following our style guidelines for hyphenated words [here](https://docs.elastic.dev/content-design/tech-docs/writing-tips#hyphenated-and-non-hyphenated-words), but am open to following a different standard if we should be following Elasticsearch's example.

+
[source,console]
--------------------------------------------------
c:\Program Files\Elastic\Agent\elastic-agent.exe restart
--------------------------------------------------

> ** **Uninstall, then re-install the {endpoint-cloud-sec} integration on affected hosts**: Uninstalling and re-installing the {endpoint-cloud-sec} integration on affected hosts will also force the uninstallation and re-installation of {elastic-endpoint} on these hosts.

Author: ```suggestion
** **Uninstall, then re-install the {endpoint-sec} integration on affected hosts**: Uninstalling and re-installing the {endpoint-sec} integration on affected hosts will also force the uninstallation and re-installation of {elastic-endpoint} on these hosts.
```

+

> NOTE: Uninstalling the {endpoint-cloud-sec} integration may temporarily cause {agent}'s status to be `Unhealthy`. The status will change to `Healthy` once the integration is re-installed.

Author: ```suggestion
NOTE: Uninstalling the {elastic-endpoint} integration may temporarily cause {agent}'s status to be `Unhealthy`. The status will change to `Healthy` once the integration is re-installed.
```

Reviewer 3: @nastasha-solomon I think this should be `{endpoint-sec}`, not `{elastic-endpoint}`, because this is referring to the integration name, not the software component

Author: Oops, you're right. I used the wrong attribute here. Thanks for catching this!

> NOTE: Uninstalling the {endpoint-cloud-sec} integration may put {agent}  in an `Unhealthy` state. This is temporary and the state will change to `Healthy` once the integration is re-installed.

Author: ```suggestion
NOTE: Uninstalling the {endpoint-sec} integration may put {agent}  in an `Unhealthy` state. This is temporary and the state will change to `Healthy` once the integration is re-installed.
```

> NOTE: Uninstalling the {endpoint-cloud-sec} integration may put {agent}  in an `Unhealthy` state. This is temporary and the state will change to `Healthy` once the integration is re-installed.

Reviewer 2: Alt suggestion since in the Endpoints list  UI, it's `agent status`. 
```suggestion
NOTE: Uninstalling the {endpoint-cloud-sec} integration may temporarily cause {agent}'s status to be  `Unhealthy`. The status will change to `Healthy` once the integration is re-installed.
```

-----------                      -- ---------------- -------
9/22/2022 10:47:35 AM          3004 Error            Windows is unable to verify the image integrity of the file \Device\HarddiskVolume3\Program Files\Elastic\Endpoint\elastic-endpo...
9/19/2022 2:10:14 PM           3004 Error            Windows is unable to verify the image integrity of the file \Device\HarddiskVolume3\Program Files\Elastic\Endpoint\elastic-endpo...
--------------------------------------------------

+

If {elastic-endpoint} is not running, there are several workarounds you can take:

** **Trigger an Endpoint reinstall by manually uninstalling {elastic-endpoint} on affected hosts**: Remove an invalid {elastic-endpoint} install by manually invoking the {elastic-endpoint} <<uninstall-endpoint,uninstall command>> on affected hosts. Once the uninstallation process has finished, use the command below to restart {agent}. Restarting {agent} automatically begins {elastic-endpoint}'s re-installation process:
+
[source,console]
--------------------------------------------------
c:\Program Files\Elastic\Agent\elastic-agent.exe restart
--------------------------------------------------

> ** **Uninstall, then re-install the {endpoint-cloud-sec} integration on affected hosts**: Uninstalling and re-installing the {endpoint-cloud-sec} integration on affected hosts will also force the uninstallation and re-installation of {elastic-endpoint} on these hosts.

Author: ```suggestion
** **Uninstall, then re-install the {endpoint-sec} integration on affected hosts**: Uninstalling and re-installing the {endpoint-sec} integration on affected hosts will also force the uninstallation and re-installation of {elastic-endpoint} on these hosts.
```

+

> NOTE: Uninstalling the {endpoint-cloud-sec} integration may temporarily cause {agent}'s status to be `Unhealthy`. The status will change to `Healthy` once the integration is re-installed.

Author: ```suggestion
NOTE: Uninstalling the {elastic-endpoint} integration may temporarily cause {agent}'s status to be `Unhealthy`. The status will change to `Healthy` once the integration is re-installed.
```

Reviewer 3: @nastasha-solomon I think this should be `{endpoint-sec}`, not `{elastic-endpoint}`, because this is referring to the integration name, not the software component

> NOTE: Uninstalling the {endpoint-cloud-sec} integration may put {agent}  in an `Unhealthy` state. This is temporary and the state will change to `Healthy` once the integration is re-installed.

Author: ```suggestion
NOTE: Uninstalling the {endpoint-sec} integration may put {agent}  in an `Unhealthy` state. This is temporary and the state will change to `Healthy` once the integration is re-installed.
```
